lib/question.js

```diff
@@ -1035,64 +801,61 @@ module.exports = {
             data.format_errors,
             data.gradable,
             data.broken,
+            // The `test` phase is not allowed to mutate `correct_answers`
+            // (aliased here to `true_answer`), so we just pick the original
+            // `true_answer` so we can use our standard `submissions_insert`
+            // sproc.
+            variant.true_answer,
             true, // regradable
             null, // credit
             null, // mode
             variant.id,
             authn_user_id,
           ];
-          sqldb.callWithClientOneRow(client, 'submissions_insert', params, (err, result) => {
+          sqldb.callOneRow('submissions_insert', params, (err, result) => {
             if (ERR(err, callback)) return;
             submission_id = result.rows[0].submission_id;
-            debug('_createTestSubmissionWithClient()', 'inserted', 'submission_id:', submission_id);
+            debug('_createTestSubmission()', 'inserted', 'submission_id:', submission_id);
+            callback(null);
+          });
+        },
+        (callback) => {
+          const params = [submission_id, authn_user_id];
+          sqldb.callOneRow('grading_jobs_insert', params, (err, result) => {
+            if (ERR(err, callback)) return;
+            grading_job = result.rows[0];
+            debug('_createTestSubmission()', 'inserted', 'grading_job_id:', grading_job.id);
             callback(null);
           });
         },
         (callback) => {
           const params = [
-            submission_id,
-            authn_user_id,
+            grading_job.id,
+            null, // received_time
+            null, // start_time
+            null, // finish_tim
+            {}, // submitted_answer
+            data.format_errors,
             data.gradable,
             data.broken,
-            data.format_errors,
-            data.partial_scores,
-            data.score,
-            data.feedback,
-            {}, // submitted_answer
             data.params,
             data.true_answer,
+            data.feedback,
+            data.partial_scores,
+            data.score,
+            null, // v2_score
           ];
-          sqldb.callWithClientOneRow(client, 'grading_jobs_insert', params, (err, result) => {
+          sqldb.callOneRow('grading_jobs_update_after_grading', params, (err, result) => {
```

Comment:
In light of #5671 and #5687, is this line also flipped from what it should be? @nwalters512

Reply:
I get it now. `grading_jobs_update_after_grading` didn't exist until this PR, I presume as a split out of `grading_jobs_insert`